src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -15001,35 +15001,204 @@ def stack(
+                ]
+            else:
+                index_cols = index_names  # type: ignore
+
+            pivot_cols = [index_cols[lev] for lev in level]  # type: ignore
+            index_cols = [
+                index_cols[i]
+                for i in range(len(index_cols))
+                if index_cols[i] not in pivot_cols
+            ]
+            vals = [
+                c for c in self.columns if c not in index_cols and c not in pivot_cols
+            ]
+            qc = self.reset_index().pivot_table(
+                columns=pivot_cols,
+                index=index_cols,
+                values=vals,
+                aggfunc="min",
+                fill_value=None,
+                margins=False,
+                dropna=True,
+                margins_name="All",
+                observed=False,
+                sort=sort,
+            )
+
+            # Set the correct index names based on if index name is None
+            output_index_names = qc.get_index_names()
+            output_index_names_replace_level_with_none = [
+                None
+                if isinstance(output_index_names[i], str)
+                and output_index_names[i][0:6] == "level_"  # type: ignore
+                else output_index_names[i]
+                for i in range(len(output_index_names))
+            ]
+            if None in output_index_names_replace_level_with_none:
+                qc = qc.set_index_names(output_index_names_replace_level_with_none)
+            # Set the correct columns names based on if column name is None
+            output_column_names = qc.columns.names
+            output_column_names_replace_level_with_none = [
+                None
+                if isinstance(output_column_names[i], str)
+                and output_column_names[i][0:6] == "level_"
+                else output_column_names[i]
+                for i in range(len(output_column_names))
+            ]
+            if None in output_column_names_replace_level_with_none:
+                from snowflake.snowpark.modin.pandas.dataframe import DataFrame
+
+                qc = (
+                    DataFrame(query_compiler=qc)
+                    .rename_axis(output_column_names_replace_level_with_none, axis=1)
+                    ._query_compiler
+                )
+        else:
+            # N.B. normally non-MultiIndex cases would throw an error in pandas
+            # for series objects. However, pandas seems to handle this for DFs
+            # by doing a transposed stack! Strange behavior, but since we throw
+            # an error in the modin API layer for Series without MultiIndex indices,
+            # we can assume that this must be a DataFrame. This assumption might
+            # change in the future, so this code should be changed along with it.
+            if has_index_names_none:
+                # Index name defaults to "index" after reset_index() operation if index name is None
```

Comment:
is the behavior of unstack is the same as stack() when it is single index for dataframe? if yes, can we just call stack here directly?

Reply:
No there are some similarities, but we don't use `sort_rows_by_column_values` and the index behavior is different.